reporter/config.go

```diff
@@ -51,4 +52,25 @@ type Config struct {
 
 	// gRPCInterceptor is the client gRPC interceptor, e.g., for sending gRPC metadata.
 	GRPCClientInterceptor grpc.UnaryClientInterceptor
+
+	// ExtraSampleAttrProd is an optional hook point for adding custom
+	// attributes to samples.
+	ExtraSampleAttrProd SampleAttrProducer
+}
+
+// SampleAttrProducer provides a hook point to:
+//
+// - inspect each trace and its meta when it is enqueued in the reporter
+// - produce extra meta info
+// - attach extra attributes to the trace
+type SampleAttrProducer interface {
+	// CollectExtraSampleMeta gathers extra sample meta-info and returns it as
+	// a pointer to a **hashable** struct.
+	CollectExtraSampleMeta(trace *libpf.Trace, meta *TraceEventMeta) any
```

Comment:
Can the returned `any` replaced with a (generic) type?
Maybe generic type parameters could help here? My worry is, that `any` might introduce subtle bugs by not using this API correctly.

Reply:
Not elegantly. It'd require us to also propagate the type parameter to `Config` and `OTLPReporter` structs.

Reply:
My personal take is, that `any`/`interfacer{}` should be avoided as return values in part of public APIs. As things are moving and this API is not considered stable, I think we can take this as an opportunity and improve in a next step.

Reply:
I generally agree with avoiding `any`, but in this case I don't think that there's any elegant way to use generics here. There's no covariance for different instances of a generic interface. That is, we can't e.g. take `SampleAttrProducer[any]` as an argument and then fill it with a `SampleAttrProducer[SomethingConcrete]`. So the generic argument will be "infectious" and propagate up. Having potentially multiple generic arguments on both the config and the reporter type seems worse than having a very localized `any`.